Adds an about section to the Circuit Maintenance Dashboard.

jvanderaa opened the pull request:
<!--
    Thank you for your interest in contributing to Circuit Maintenance! Please note
    that our contribution policy recommends that a feature request or bug
    report be opened for approval prior to filing a pull request. This
    helps avoid wasting time and effort on something that we might not
    be able to accept.

    Please indicate the relevant feature request or bug report below.
-->

# Closes: #233

## What's Changed

<!--
    Please include:
    - A summary of the proposed changes
    - A sectioned breakdown for larger features under ## subheadings
    - Screenshots, example payloads where relevant:
      - Before/After for bugfixes
      - Using a new feature
-->

Adds an **About** card to the Circuit Maintenance dashboard, built with Nautobot UI
constructs (`card` / `card-header` / `card-body`, the `placeholder` filter, and `<code>`
formatting for version values). The card surfaces two derived values, both read from
package metadata (nothing hardcoded):

- **App Version** — the installed `nautobot-circuit-maintenance` version.
- **Circuit Maintenance Parser Version** — the installed `circuit-maintenance-parser` version.

This is useful for verifying which version of the app is deployed and which version of
the parser logic is processing notifications.

`develop` has been merged in to keep the branch current (Nautobot 3.x / Django 5.2).

### Screenshots

| Light | Dark |
| --- | --- |
| ![About card - light](https://github.com/nautobot/nautobot-app-circuit-maintenance/blob/233_about_page/docs/images/about_card_light.png?raw=true) | ![About card - dark](https://github.com/nautobot/nautobot-app-circuit-maintenance/blob/233_about_page/docs/images/about_card_dark.png?raw=true) |

## To Do

<!--
    Please feel free to update todos to keep track of your own notes for WIP PRs.
-->
- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/core/#creating-changelog-fragments))
- [x] Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [x] Documentation Updates (when adding/changing features)
- [x] Outline Remaining Work, Constraints from Design

jvanderaa committed:
Adds an about section to the Circuit Maintenance Dashboard.

jvanderaa committed:
Adds Circuit Maintenance Parser version to the dashboard.

glennmatthews commented on nautobot_circuit_maintenance/templates/nautobot_circuit_maintenance/circuit_maintenance_overview.html:
```diff
@@ -119,6 +119,19 @@ <h3 class="text-start">Metrics</h3>
         </div>
+        <div class="card">
+            <div class="container">
+                <h3 class="text-start">About</h3>
```
Might be more aesthetic to do a `div class="card-header"` in place of the `h3`?

jvanderaa replied:
<img width="408" height="826" alt="image" src="https://github.com/user-attachments/assets/c35faa3c-3303-4508-9056-ac76078f3468" />
If we do the card-header then we should probably do it in the rest of the sections too. Thoughts @glennmatthews ?

glennmatthews replied:
I'd be in favor of that, yes. The current styling is very inconsistent with the rest of Nautobot, IMHO.

jvanderaa replied:
I'm good by that. I'll be diving in later today looking for a styling guide or how I can make the dashboard page look more appropriate.

jvanderaa replied:
Done — I reworked all four dashboard sections (Circuits, Devices, Metrics, and the new About card) to use `<div class="card-header">` for consistency with the rest of Nautobot, rather than the old `h3` styling. Should be good for another look @glennmatthews.

jvanderaa committed:
Add integration and unmocked tests for dashboard About card

jvanderaa committed:
Refactor dashboard card titles to use card-header for Nautobot consistency

jvanderaa committed:
Build out dashboard About card with app and parser versions

Refactor the About card to use Nautobot card-body construct and the
placeholder filter, and add the app version alongside the Circuit
Maintenance Parser version (both derived from package metadata).

jvanderaa committed:
Merge branch 'develop' into 233_about_page

# Conflicts:
#	nautobot_circuit_maintenance/templates/nautobot_circuit_maintenance/circuit_maintenance_overview.html

jvanderaa committed:
Document dashboard About card with app version and screenshots

Update the About card docs to note it now shows the app version in
addition to the parser version, and add light/dark screenshots.
